go-controller/pkg/ovn/endpoints.go

```diff
@@ -107,12 +108,26 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 						continue
 					}
 				}
+				ovn.handleExternalIPs(svc, svcPort, ips, targetPort)
 			}
 		}
 	}
 	return nil
 }
 
+func (ovn *Controller) handleExternalIPs(svc *kapi.Service, svcPort kapi.ServicePort, ips []string, targetPort int32) {
+	logrus.Infof("handling external IPs for svc %v", svc.Name)
+	if len(svc.Spec.ExternalIPs) == 0 {
+		return
+	}
+	for _, extIP := range svc.Spec.ExternalIPs {
+		err := ovn.createLoadBalancerVIP(ovn.getDefaultGatewayLoadBalancer(svcPort.Protocol), extIP, svcPort.Port, ips, targetPort)
```

Comment:
ovn.getDefaultGatewayLoadBalancer should be called in a different line and we should "continue" if return is empty. Otherwise, we will likely have a problem.